Feature/phase01

AmirRezaZahedi opened the pull request.

AmirRezaZahedi committed:
Class defined and json read

AmirRezaZahedi committed:
Create README.md

AmirRezaZahedi committed:
we can calculate student's average

AmirRezaZahedi committed:
Merge branch 'feature/phase01' of https://github.com/Star-Academy/Summer1403-SE-Team01 into feature/phase01

AmirRezaZahedi committed:
finish with different implement:)

AmirRezaZahedi committed:
last changing

AmirRezaZahedi committed:
Delete additonal file:)

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+        {
+            get { return _lastName; }
+            set { _lastName = value; }
+        }
```
public int StudentNumber { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    
    اینجوری خوانا تره و تمیز تره

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+    }
+
+
+    public class StudentScore
```
این کلاسم با get;set مانند مورد بالا بهبود پیدا کنه

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+                PropertyNameCaseInsensitive = true
+            };
+
+            var studentsJson = File.ReadAllText("./files/students.json");
```
duplicate code

AliSK81 replied:
move file paths to resource (.resx) file

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+            List<Student> students = JsonSerializer.Deserialize<List<Student>>(studentsJson, _options);
+
+            var scoreJson = File.ReadAllText("./files/scores.json");
+            List<StudentScore> studentScores = JsonSerializer.Deserialize<List<StudentScore>>(scoreJson, _options);
```
duplicate code

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+
+            var scoreJson = File.ReadAllText("./files/scores.json");
+            List<StudentScore> studentScores = JsonSerializer.Deserialize<List<StudentScore>>(scoreJson, _options);
+
```
برای خواندن از جیسون یک کلاس جنریک بزنید که کار خوندن از جیسون انجام میده این کلاس میتونه یه interface پیاده سازی کنه 
چون ممکنه یه روزی نخایم دیگه از فایل بخونیم و به جاش بخایم از db بخونیم میریم اون interface رو براش پیاده سازی مجدد مینویسیم

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+                cur.Courses.Add(ss.Lesson, ss.Score);
+                sum += ss.Score;
+                count++;
+            }
```
این for ببنیید با linq میشه تمیز تر نوشت 
و اینکه بره داخل یه تابع

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+            
+            var sorted = students.OrderByDescending(x=>x.Average).Take(3);
+            foreach(var s in sorted)
+                Console.WriteLine($"name: {s.FirstName} {s.LastName}, avg: {s.Average}");
```
$"name: {s.FirstName} {s.LastName}, avg: {s.Average}" 
برا این یه کلاس بزنید و string بازنویسی کنید

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+            var studentsJson = File.ReadAllText("./files/students.json");
+            List<Student> students = JsonSerializer.Deserialize<List<Student>>(studentsJson, _options);
+
+            var scoreJson = File.ReadAllText("./files/scores.json");
```
در مورد async بخونید و سعی کنید بعد از نوشتن اون کلاسی که گفتم خواندن فایل async کنید و از متد 
ReadAllTextAsync 
استفاده کنید

tahadavari commented on project/Program.cs:
```diff
@@ -0,0 +1,111 @@
+
+namespace ScoreBoard
+{
+    public class Student
```
کلاس هاتونو در فایل های مجزا بزارید

AmirRezaZahedi replied:
done:)

tahadavari reviewed (changes requested):
یه بررسی بکنید 
دمتون گرم خسته نباشید

AmirRezaZahedi committed:
We try out to change code based on mentor's opinion

AliSK81 commented on project/bin/Debug/net8.0/project.deps.json:
محتوای تمام فولدرهای 
obj
bin
.idea
برن تو گیت ایگنور و از pr حذف بشن

AliSK81 reviewed (changes requested):
کامنت های قبلی ریزالو نشده که

AmirRezaZahedi committed:
Refactor project configuration and error handling

- Replaced JSON configuration with RESX format for better localization support.
- Implemented try-catch blocks to enhance error handling.
- Updated .gitignore to exclude IDE-specific files and build artifacts:
  - Added .idea/ directory
  - Added bin/ and obj/ directories

AmirRezaZahedi committed:
Remove bin and obj directories from version control, add try-catch, and switch to RESX format

- Removed bin/ and obj/ directories from version control.
- Added try-catch blocks for improved error handling.
- Replaced JSON configuration with RESX format for better localization support.
